service/ciphering/aesEncoder_test.go

```diff
@@ -0,0 +1,96 @@
+package ciphering
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+const isBase64 = "^(?:[A-Za-z0-9+/]{4})*(?:[A-Za-z0-9+/]{2}==|[A-Za-z0-9+/]{3}=|[A-Za-z0-9+/]{4})$"
+const encryptedString = "848d56796ac855b0b3f06cb77bb652d2a9a4b84f64026dbfbfe94aaf76cb843608c85a2112ac"
+const sourceString = "testString"
+const key = "1234567890123456"
+const wrongKey = "23"
+
+func TestAesEncoder_Encrypt_TableDriven(t *testing.T) {
+	var encryptionTestCases = []struct {
+		name      string
+		source    string
+		key       string
+		expected  string
+		expectErr bool
+	}{
+		{
+			name:      "encryption valid case",
+			source:    sourceString,
+			key:       key,
+			expected:  sourceString,
+			expectErr: false,
+		},
+		{
+			name:      "encryptiom wrong key size",
+			source:    sourceString,
+			key:       wrongKey,
+			expected:  "",
+			expectErr: true,
+		},
+	}
+
+	for _, tc := range encryptionTestCases {
+		t.Run(tc.name, func(t *testing.T) {
+			encoder := NewAESEncoder()
+			result, err := encoder.Encrypt(tc.source, []byte(tc.key))
+			if tc.expectErr {
+				require.Error(t, err)
+			} else {
+				require.Nil(t, err)
+				require.Regexp(t, isBase64, result)
```

Comment:
Should also be comparison with an expected value.

Reply:
unfortunately we cant process comparison with expected encrypted value because it is generated differently every time.

Reply:
> Should also be a comparison with an expected value.

unfortunately we cant process comparison with expected encrypted value because it is generated differently every time.

Reply:
> That's a good point and an opportunity to refactor the encoder in such a way that for tests it would become possible to predict expected result.
> 
> Do you see any option that could help up to achieve this?
Yes, i think we can create method which helds randomized part of aes encryption and mock it in the test. I'll do it soon and commit

Reply:
> That's a good point and an opportunity to refactor the encoder in such a way that for tests it would become possible to predict expected result.
> 
> Do you see any option that could help up to achieve this?

for some reasons my comment was not displayed so here are the [changes i have made ](https://github.com/retttddd/golangProjectv3/pull/2/commits/4efe8f89df6b1a2c89eaabd85d043348907654a4)